core/src/main/scala/com/softwaremill/diffx/DiffFor.scala

```diff
@@ -1,14 +1,29 @@
 package com.softwaremill.diffx
 
-trait DiffFor[T] {
-  def apply(left: T, right: T): DiffResult
+trait DiffFor[T] { outer =>
+  protected val ignored: List[List[String]] = List.empty
```

Comment:
maybe create at least a type alias for `List[String]` so that the type would be e.g. `List[FieldPath]` instead of a cryptic `List[List[String]]`?